medcat/cat.py

```diff
@@ -1081,6 +1086,20 @@ def train_supervised_raw(self,
                                                                                use_overlaps=use_overlaps,
                                                                                use_groups=use_groups,
                                                                                extra_cui_filter=extra_cui_filter)
+        if (train_meta_cats and
+                # NOTE if no annnotaitons, not point
+                count_all_annotations(data) > 0):  # type: ignore
+            # NOTE: if there
+            logger.info("Training MetaCATs within train_supervised_raw")
+            _, _, ann0 = next(iter_anns(data))  # type: ignore
+            for meta_cat in self._meta_cats:
+                # only consider meta-cats that have been defined for the category
```

Comment:
the logic that does the checking if the category can be exchanged with a different name, if thats available on `medcat.meta_cat`? @shubham-s-agarwal  - is the category checking, swapping logic callable?

Reply:
Yeah, I did have a note about this in the PR comment:
> PS:
> With https://github.com/CogStack/MedCAT/pull/515 the check for suitable MetaCATs according to category name may need adjusting since the PR allows for alterantives category names.

Reply:
Though doesn't look like it's easily callable. Would need some kind of general use category name resolver.